examples/kyb-app/DynamicElements/hooks/useUIElement/hooks/useHandlers/useHandlers.ts

```diff
@@ -0,0 +1,58 @@
+import { isEventRule } from '@app/components/organisms/DynamicElements/helpers/is-event-rule';
+import { useDynamicUIContext } from '@app/components/organisms/DynamicElements/hooks/useDynamicUIContext';
+import { Action, UIElement } from '@app/domains/collection-flow';
+import { AnyObject } from '@ballerine/ui';
+import { useMemo } from 'react';
+
+export interface ElementHandlers {
+  onClick: React.MouseEventHandler<any> | undefined;
+  onChange: React.ChangeEventHandler<any> | undefined;
+}
+
+export const useHandlers = (element: UIElement<AnyObject>, actions: Action[]) => {
+  const { dispatchActions, updateContext, getContext } = useDynamicUIContext();
+  const scopedActions = useMemo(() => {
+    const scopedActions = actions.filter(action => {
+      return Boolean(
+        action.dispatchOn.find(rule => {
+          if (isEventRule(rule)) {
+            return rule.value.uiElementName === element.name;
+          }
+          return false;
+        }),
+      );
```

Comment:
Are both `filter` and `find` needed here?
```suggestion
      return  action.dispatchOn.some(rule => isEventRule(rule) && rule.value.uiElementName === element.name);
```

Reply:
Yep,the logic is to find actions scoped to specific ui element.Actions could be linked to multiple UI Elements and vice-versa.In order to find actions specific only to current UIElement we have to run multiple searches.